src/components/UserManagement/UserShipping/UserShipping.tsx

```diff
@@ -212,78 +233,113 @@ export default class UserShipping extends React.Component<Props, State> {
                     }}
                   />
                 )}
+                {this.state.editAddressNetworkError && (
+                  <ErrorLabel
+                    className={styles.errorCreateAddress}
+                    errorText={this.state.editAddressNetworkError}
+                  />
+                )}
               </div>
             </div>
           </div>
         </div>
+
+        {this.context.customerLoading && (
+          <Loader
+            containerClassName={styles.loaderContainer}
+            loaderClassName={styles.loader}
+          />
+        )}
       </div>
     );
   }
 
-  onSaveAddress = (addressValues: AddressFormValuesT, addressId: string) => {
-    const newAddress = new Address({
-      id: addressId || Math.random(),
-      nickname: addressValues.nickname,
-      firstName: addressValues.firstName,
-      lastName: addressValues.lastName,
-      addressLine1: addressValues.addressLine1,
-      addressLine2: addressValues.addressLine2,
-      city: addressValues.city,
-      state: addressValues.state,
-      zipcode: addressValues.zipcode,
-      default: addressValues.default,
+  onSaveAddress = (addressValues: AddressFormValuesT, addressId: number) => {
+    this.setState({
+      createAddressNetworkError: "",
+      editAddressNetworkError: "",
     });
 
-    if (newAddress.default) {
-      const newAddresses = this.state.addresses.map((address) => {
-        address.default = false;
+    const addressFields = {
+      company: addressValues.nickname,
+      firstName: addressValues.firstName || "",
+      lastName: addressValues.lastName || "",
+      city: addressValues.city || "",
+      region: addressValues.state || "",
+      address: addressValues.addressLine1 || [] || "",
```

Comment:
I don't think we should be using `[]` as a default here or for the aptNumber field. the CustomerAddressInput class already constructs the array `[addressValues.address, addressValues.aptNumber]` if the fields are present. As it is now, if the aptNumber is missing, the final value ends up looking like `[addressValues.address, []]` which triggers a graphql error.